test/parallel/test-net-better-error-messages-port-hostname.js

```diff
@@ -3,12 +3,18 @@ const common = require('../common');
 const net = require('net');
 const assert = require('assert');
 
-const c = net.createConnection(common.PORT, '***');
+const s = net.createServer();
+s.listen(0);
+const port = s.address().port;
```

Comment:
@thefourtheye Might be overthinking it indeed, but I prefer a valid port because IMO selecting port `0` for `createConnection()` is undefined behavior, so I'd rather not codify the expectations in tests. It's also kind of misleading because someone reading the code might infer that it selects an arbitrary port but it doesn't. So I'd rather see a valid port, and any valid port will do, so `80` or `443` or `8080` or whatever would all be fine by me. 

All that said, you're correct that what you suggest above will work and if you feel that `0` is the way to go and @adityaanandmc agrees, I won't object. I have opinions, but not strong ones.